### ISSUE TITLE
Learner view: retry functionality for independent practice quiz is implemented

### ISSUE THREAD

**Issue by sairina** [learningequality/kolibri#8510]
### User stories for this issue:

> As a learner, I want to be able to retake/retry an independent practice quiz from a channel OR from a lesson so that I can improve my score.

(Please note this issue is for both retrying a practice quiz from a channel and lesson)

|Frame|Explanation|
|--|--|
|![Screen Shot 2021-10-20 at 8 25 03 PM](https://user-images.githubusercontent.com/13563002/138210913-302a26eb-973d-4273-a97d-07ec17e74e41.png)|Learner navigates to independent practice quiz report from lesson or channel and clicks on "Practice again" button (see Figma for details)|
|![Screen Shot 2021-10-20 at 8 54 28 PM](https://user-images.githubusercontent.com/13563002/138210987-0313c99a-7c6d-4e69-853b-79ba91587911.png)|Learner is directed to practice quiz|
|![Screen Shot 2021-10-20 at 8 55 22 PM](https://user-images.githubusercontent.com/13563002/138211064-cf633ac1-bf90-44a1-8c7b-6f1e2a22f343.png)| **IF CHANNEL**: Learner is returned to channel and sees the completion data (score) for the most recent attempt on content card|
|![Screen Shot 2021-10-20 at 8 24 40 PM](https://user-images.githubusercontent.com/13563002/138211356-f3d255b2-3bf8-4a55-b86e-b46495b842e9.png)| **IF LESSON**: Learner is returned to lesson and sees the completion data (score) for the most recent attempt on card|
|![Screen Shot 2021-10-20 at 8 25 03 PM](https://user-images.githubusercontent.com/13563002/138211406-a73e03ef-9134-4bae-a5e5-9c33f7b92cab.png)| **IF FIRST ATTEMPT**: Learner clicks on the content card from frame above and sees this view of the report|
|![Screen Shot 2021-10-20 at 9 26 21 PM](https://user-images.githubusercontent.com/13563002/138211497-6ad87896-5975-4619-bf3e-216f76e4fdd1.png)| **IF AFTER FIRST ATTEMPT**: Learner clicks on content card from frame above and sees this view of the report|

### Notes and changes to current designs:
- [Figma for retry from a **lesson**](https://www.figma.com/file/TfaSTT9gs6fPKhGdNUjffU/Channel-quizzes?node-id=919%3A6694)
- [Figma for retry from a **channel**](https://www.figma.com/file/TfaSTT9gs6fPKhGdNUjffU/Channel-quizzes?node-id=911%3A6510)
- This is to address what happens after a learner clicks on the "Practice again" button in their report
- Behavior for the retry functionality is the same for when learner access this function either from a lesson or channel; however, routing is different
- Routing and navigation must be created to and from channel, and to and from lesson
- All strings required should already be in `commonCoreStrings`, `commonCoachStrings`, or `commonLearnStrings`
- Important to note that the `ReportsPracticeQuizLearnerPage` will differ depending on whether this was the learner's first attempt, or after multiple (see frames above)
- If a learner exits a practice quiz before they have submitted,
   - the timer resets
   - user will be returned to the page where they found the independent practice quiz (channel or lesson)
   - the "Completion status" at the bottom of the card should reset to "in progress"

### Criteria for success:
**Please copy these into your PR for "Steps for Testing"!**

_This issue is done when the checkbox and Gherkin story below works_

- [ ] Learners will see a different report depending whether learner it has been the first attempt on the practice quiz or it has been multiple
- [ ] Learners will see the most recent attempt's score on their lesson or channel content card after submitting quiz
- [ ] Learners can retry an independent practice quiz from a channel or lesson
- [ ] If a learner exits a practice quiz before they have submitted,
   - [ ] the timer resets
   - [ ] user will be returned to the page where they found the independent practice quiz (channel or lesson)
   - [ ] the "Completion status" in the content card should reset to "in progress"
```
Feature: Learners can retry an independent practice quiz
	Scenario: Learner can retake an independent practice quiz that was assigned in a lesson or found in a channel
			Given that I have finished and submitted the independent practice quiz
				And I am back on in the channels tab (or, for a lesson, back in my lesson)
				And I see my score on the same independent practice quiz content card
				And I click on the content card
				And I see my independent practice quiz report
			When I click on the button to retake the independent practice quiz
				Then I see that same independent practice quiz appear, ready to be taken and submitted
```

**Comment by bjester** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Kolibri-Ecosystem-6047e525b8b9d900182e2d6f/issues/learningequality/kolibri/8510?planning-poker) with ZenHub @jredrejo @marcellamaki @sairina @rtibbles @MisRob @nucleogenesis @jamalex

**Comment by sairina** [comment]
From the sprint planning meeting, want to note that this issue, from the frontend perspective, will require the use of sending the `repeated` flag to the backend.

**Comment by rtibbles** [comment]
Fixed in #8645
